zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -666,13 +674,38 @@ impl<'a, F: Field> EVMConstraintBuilder<'a, F> {
         self.opcode_lookup_at_rlc(index, opcode, 0.expr());
     }
 
+    #[cfg(feature = "dual_bytecode")]
+    pub(crate) fn opcode_lookup2_at(
+        &mut self,
+        index: Expression<F>,
+        opcode: Expression<F>,
+        is_code: Expression<F>,
+    ) {
+        assert_eq!(is_code, 1.expr());
+        self.opcode_lookup_at_rlc2(index, opcode, 0.expr());
+    }
+
     pub(crate) fn opcode_lookup_rlc(&mut self, opcode: Expression<F>, push_rlc: Expression<F>) {
         self.opcode_lookup_at_rlc(
-            self.curr.state.program_counter.expr() + self.program_counter_offset.expr(),
+            self.curr.state.program_counter.expr() + self.program_counter_offset.clone(),
             opcode,
             push_rlc,
         );
-        self.program_counter_offset += 1;
+        // self.program_counter_offset += 1;
+        self.program_counter_offset = self.program_counter_offset.clone() * self.condition_expr();
```

Comment:
not sure why need to change the `self.program_counter_offset += 1;`?

Reply:
now this method is in `cb.condition` now.  only condition is true,  then plus `program_counter_offset`.